# @rjsf/coreの newKey部分のtitleに文字列を割り当てられるようにする
diff --git a/node_modules/@rjsf/core/dist/core.umd.js b/node_modules/@rjsf/core/dist/core.umd.js
index c6d0797..6b954fa 100644
--- a/node_modules/@rjsf/core/dist/core.umd.js
+++ b/node_modules/@rjsf/core/dist/core.umd.js
@@ -2033,7 +2033,7 @@
         return;
       }
       const newFormData = { ...formData };
-      const newKey = getAvailableKey("newKey", newFormData);
+      const newKey = getAvailableKey(utils.getUiOptions(uiSchema, globalUiOptions).newKeyName ?? "newKey", newFormData);
       if (schema.patternProperties) {
         utils.setByPath(newFormData, newKey, null);
       } else {
diff --git a/node_modules/@rjsf/core/dist/index.cjs b/node_modules/@rjsf/core/dist/index.cjs
index f65e413..5965763 100644
--- a/node_modules/@rjsf/core/dist/index.cjs
+++ b/node_modules/@rjsf/core/dist/index.cjs
@@ -2135,7 +2135,7 @@ function ObjectField(props) {
       return;
     }
     const newFormData = { ...formData };
-    const newKey = getAvailableKey("newKey", newFormData);
+    const newKey = getAvailableKey((0, import_utils11.getUiOptions)(uiSchema, globalUiOptions).newKeyName ?? "newKey", newFormData);
     if (schema.patternProperties) {
       (0, import_utils11.setByPath)(newFormData, newKey, null);
     } else {
diff --git a/node_modules/@rjsf/core/dist/index.esm.js b/node_modules/@rjsf/core/dist/index.esm.js
index 90fbfc5..7e4efc9 100644
--- a/node_modules/@rjsf/core/dist/index.esm.js
+++ b/node_modules/@rjsf/core/dist/index.esm.js
@@ -2223,7 +2223,7 @@ function ObjectField(props) {
       return;
     }
     const newFormData = { ...formData };
-    const newKey = getAvailableKey("newKey", newFormData);
+    const newKey = getAvailableKey(getUiOptions10(uiSchema, globalUiOptions).newKeyName ?? "newKey", newFormData);
     if (schema.patternProperties) {
       setByPath3(newFormData, newKey, null);
     } else {
diff --git a/node_modules/@rjsf/core/lib/components/fields/ObjectField.js b/node_modules/@rjsf/core/lib/components/fields/ObjectField.js
index ec254fe..59606b6 100644
--- a/node_modules/@rjsf/core/lib/components/fields/ObjectField.js
+++ b/node_modules/@rjsf/core/lib/components/fields/ObjectField.js
@@ -144,7 +144,7 @@ export default function ObjectField(props) {
             return;
         }
         const newFormData = { ...formData };
-        const newKey = getAvailableKey('newKey', newFormData);
+        const newKey = getAvailableKey(getUiOptions(uiSchema, globalUiOptions).newKeyName ?? 'newKey', newFormData);
         if (schema.patternProperties) {
             setByPath(newFormData, newKey, null);
         }
diff --git a/node_modules/@rjsf/core/src/components/fields/ObjectField.tsx b/node_modules/@rjsf/core/src/components/fields/ObjectField.tsx
index bb009b8..ca94d04 100644
--- a/node_modules/@rjsf/core/src/components/fields/ObjectField.tsx
+++ b/node_modules/@rjsf/core/src/components/fields/ObjectField.tsx
@@ -293,7 +293,7 @@ export default function ObjectField<T = any, S extends StrictRJSFSchema = RJSFSc
       return;
     }
     const newFormData = { ...formData } as T;
-    const newKey = getAvailableKey('newKey', newFormData);
+    const newKey = getAvailableKey((getUiOptions(uiSchema, globalUiOptions).newKeyName as string | undefined) ?? 'newKey', newFormData);
     if (schema.patternProperties) {
       setByPath(newFormData, newKey, null);
     } else {
